ls/src/main.rs

```diff
@@ -0,0 +1,351 @@
+use std::{
+    fs,
+    io::{self, Write},
+    os::unix::fs::MetadataExt,
+    path, process,
+    string::String,
+    time::SystemTime,
+};
+
+use pad::{Alignment, PadStr};
+
+extern crate chrono;
+
+mod cli;
+mod file;
+mod flags;
+
+use file::File;
+use flags::Flags;
+
+fn main() -> io::Result<()> {
+    let matches = cli::create_app().get_matches();
+
+    let files = matches.values_of("FILE").unwrap();
+    let flags = Flags::from_matches(&matches);
+
+    let mut exit_code = 0;
+
+    let mut writer: Box<dyn Write> = Box::new(io::stdout());
```

Comment:
I think it doesn't need to be a dynamic dispatched `Write`, but I do think it would be better to use `BufWritter` to have better performance writing